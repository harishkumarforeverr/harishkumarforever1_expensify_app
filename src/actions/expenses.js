import { v4 as uuid } from 'uuid';
import database from "../firebase/firebase";
// ADD_EXPENSE
export const addExpense = (expense) => ({
    type: "ADD_EXPENSE",
    expense
});
//StartAddExpense
export const StartAddExpense=(expense={})=>{
    return (dispatch)=>{
          const { description = "",  note = "", amount = 0, createdAt = 0  }=expense
          const addingExpensesObject={description,note,amount,createdAt };
          return database.ref("expenses").push(addingExpensesObject).then((snapshot)=>{
              dispatch(addExpense({
                  id:snapshot.key,
                  ...addingExpensesObject
              }))
          });
    }
}

// REMOVE_EXPENSE

export const removeExpense = ({ id } = {}) => ({
    type: "REMOVE_EXPENSE",
    id
});
// EDIT_EXPENSE

export const editExpanse = (id, update) => ({
    type: "EDIT_EXPANSE",
    id,
    update
})

//SET expenses
export const setExpenses=(expense)=>({
    type:"SET_EXPENSES",
    expense 
})

export const startSetExpenses=()=>{
    return (dispatch)=>{
        return database.ref("expenses").once("value").then((snapshot)=>{   
            const expenses=[];
            snapshot.forEach((childSnapShot)=>{     
                expenses.push({
                    id:childSnapShot.key, 
                    ...childSnapShot.val()
              })
            })
            dispatch(setExpenses(expenses));
        })
    }
}