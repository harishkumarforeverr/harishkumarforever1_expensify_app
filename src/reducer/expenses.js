
// expanse Reducer 
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {  
        case "ADD_EXPENSE":
            return ([
                ...state,
                action.expense
            ]);
        case "REMOVE_EXPENSE":
            return state.filter((expense) => expense.id !== action.id);
        case "EDIT_EXPANSE" :
            return state.map((expanse)=>{
               if(expanse.id===action.id){
                   return {
                    ...expanse,
                    ...action.update
                   }
               }
               else{
                   return expanse;
               }
            });
        case "SET_EXPENSES":
            return action.expense; 
        default: return state
    }
}
