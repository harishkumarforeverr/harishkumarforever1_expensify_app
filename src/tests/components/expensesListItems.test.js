import React from "react";
import {shallow} from "enzyme";
import ExpensesListItems from "../../components/expensesListItems";
import expenses from "../fixtures/expenses";
 
test("test/componnets/expensesListItems  should render expensesListItems correctly",()=>{  
        const wrapper=shallow(<ExpensesListItems {...expenses[1]}/>);
        expect(wrapper).toMatchSnapshot();
});