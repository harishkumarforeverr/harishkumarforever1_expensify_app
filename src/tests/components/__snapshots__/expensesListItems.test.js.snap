// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/componnets/expensesListItems  should render expensesListItems correctly 1`] = `
<div>
  <Link
    to="/edit/1"
  >
     
    <h3
      className="expenses"
    >
         
      my room rent
       
    </h3>
     
  </Link>
  <p>
    $25.0000
    --
    December 28th, 1969
  </p>
  <hr />
</div>
`;
